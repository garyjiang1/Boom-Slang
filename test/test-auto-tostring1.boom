class MyGeneric[T]:
	static:
		T static1 = 0
		string static2 = "hey"

	required:
		float x

	optional:
		char c = 'h'
		int[][] arr = default int[2][5]

class MyClass = MyGeneric(int)
MyClass instance = MyClass(2.0)
println(instance)

MyClass instance2 = NULL
println(instance2)
