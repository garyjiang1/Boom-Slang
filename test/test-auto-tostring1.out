MyClass:
static1:0
static2:hey
x:2.0000
c:h
arr:Array

NULL
